lib/project_types/script/layers/infrastructure/languages/task_runner.rb

```diff
@@ -5,15 +5,35 @@ module Layers
     module Infrastructure
       module Languages
         class TaskRunner
```

Comment:
I'm OK with the approach of defining a base `TaskRunner` class, but I will offer another way to think about this for your consideration.  The way you have it here, we are saying: `ASTaskRunner`, `TSTaskRunner`, and `WasmTaskRunner` are all `TaskRunner`s.  However, with Wasm being the default and foundational, another way to think about is that `WasmTaskRunner` is the default, and `ASTaskRunner` and `TSTaskRunner` are specialized `WasmTaskRunner`, and `WasmTaskRunner` contains the basics that all other children need.  Currently, there are no dependencies to install in `WasmTaskRunner`, but once we introduce the need for setting up Wasm runtime (RuntimeEngine CLI) to run scripts locally, then that dependency will apply to other TaskRunners.  We can leave it as you have it here, and wait to see how this design evolves.  I just wanted to say that, if there is ever a case where some logic becomes common to all TaskRunners because of Wasm, it's probably time to re-evaluate the relationships here.  Hypothetically when that happens, this `self.for` doesn't make sense in `WasmTaskRunner` any more, and we should probably just have a separate Factory class that returns the appropriate `TaskRunner` and `ProjectCreator` per language.

Reply:
This line of thinking makes a lot of sense. I do think that we should wait until changing it becomes necessary so that we can see how things evolve from now until then.